tedana/workflows/post_taskcorr.py

```diff
@@ -0,0 +1,259 @@
+            LGR.warning('Wrong orientation, transposing the design matrix.')
+            ts = ts.T
+
+    return ts
+
+
+def correlate_ts(ts, mixmat, thr=0.6):
+    """
+    Run (normalised) cross correlation of the signals
+    And then selects maximum NCC to decide which one to flag.
+    Also threshold components and then save their index.
+
+    Parameters
+    ----------
+    ts: T x N numpy.array
+        Timeseries to compare components to.
+    mixmat: T x N numpy.array
+        Timeseries of MEICA components.
+    thr: float, Optional
+        Threshold for NCC value considered significant. Default = 0.6.
+
+    Returns
+    -------
+    selcomp: list
+        List of indexes of components that are significantly similar to any of the N
+        timeseries of ts.
+
+    """
+    _, ncomp = mixmat.shape
+
+    if np.size(ts.shape) != 1:
+        _, nts = ts.shape
+    else:
+        nts = 1
+
+    corr_mtx = np.empty((ncomp, nts))
+
+    for ts_num in range(0, nts):
+        for comp_num in range(0, ncomp):
+            ncc = np.correlate(mixmat[:, comp_num], ts[:, ts_num], mode='same')
+            corr_mtx[comp_num, ts_num] = np.max(ncc)
+
+    thr_mtx = corr_mtx > thr
+    mix_mask = thr_mtx.any(axis=1)
+    selcomp = [i for i, x in enumerate(mix_mask) if x]
+
+    return selcomp
+
+
+def modify_comp_table(ctab_fullpath, selcomp, flag='accepted'):
+    """
+    Function to change flag of components in the components table.
+
+    Parameters
+    ----------
+    ctab_fullpath: string
+        Full path to components table as of tedana output.
+        !!! It will be overwritten !!!
+    selcomp: list
+        List of indexes of components to be flagged with flag.
+    flag: string, Optional
+        Flag to change the label of the components with index
+        selcomp, in the components table.
+
+    Note
+    ----
+    The file output is the same ctab_fullpath
+
+    """
+
+    comptable = pd.read_csv(ctab_fullpath, sep='\t', index_col='component')
+
+    comptable['original_classification'] = comptable['classification']
+    comptable['original_rationale'] = comptable['rationale']
+    LGR.info('Overriding original classifications with correlation with'
+             'input timeseries (%s components)', flag)
+    comptable.loc[selcomp, 'classification'] = flag
+    if flag == 'accepted':
+        comptable.loc[selcomp, 'rationale'] = 'I099'
+    else:
+        comptable.loc[selcomp, 'rationale'] = 'I098'
+
+    LGR.info('Overwriting original component table.')
+    comptable.to_csv(ctab_fullpath, sep='\t', index_label='component')
+
+
+def check_task_corr(mixmat, ctab_fullpath, ts, thr=0.6, flag='accepted'):
+    """
+    This function is the workflow of the component selection.
+
+    Parameters
+    ----------
+    mixmat: T x N :obj:`numpy.array`
+        Timeseries of MEICA components.
+    ctab_fullpath: :obj:`string`
+        Full path to components table as of tedana output.
+        !!! It will be overwritten !!!
```

Comment:
Sorry to just jump in here but maybe it would be nice to make a backup copy of the original components table before overwriting it? Just adding in a line like `shutil.copy(ctab_fullpath, ctab_fullpath + '.bak')` prior to saving the modified table would be nice!

If this modifies things in a way that a user deems "not great" then they can recover the original comptable without too much hassle rather than having to re-run most of the `tedana` workflow.

Reply:
The original classifications and rationales will be retained in "original_classification" and "original_rationale" columns though.

Reply:
But if I wanted to use e.g., the tools in `tedana.viz`, those primarily pulls from the "classification" column, right? I guess I can go in and just over-write 'original_classification' to 'classification' if I don't like the `taskcorr` output...

Alright, sorry about that! :+1:

Reply:
I think it's pretty easy to fix _now_ (since we use csvs), but once we switch to json files for the decompositions it will probably be a good idea to come back to this and to come up with a programmatic solution.

Reply:
@rmarkello , I might like your idea more.
@tsalo , @emdupre , what do you think?